src/proposal.mligo

```diff
@@ -356,7 +439,7 @@ let handle_proposal_is_over
       && counter.current < counter.total // not finished
     then
       let counter = { counter with current = counter.current + 1n } in
-      let cond =    do_total_vote_meet_quorum_threshold(proposal, store, config.quorum_threshold)
+      let cond =    do_total_vote_meet_quorum_threshold(proposal, store, store.quorum_threshold_at_stage.quorum_threshold)
                 && proposal.upvotes > proposal.downvotes
```

Comment:
This function (`handle_proposal_is_over`) is used by `flush`, but I don't see `update_quorum` be called in the latter, should't it be?

Reply:
Right now it is only called during `propose` call.  Do you think that might not be enough. May be I have missed something,

Reply:
Yeah I think `flush` needs update the quorum as well, because there may be cases in which a `propose` wasn't called yet.

Take the case in which one `propose`s in a stage, `vote`s in the next and waits for this stage to end.
It's possible that now we are again in a `proposing` period and the proposal above can be `flush`ed.
If `flush` is called before `propose` than the quorum would be outdated and remain the one for the previous cycle.

Reply:
> If flush is called before propose than the quorum would be outdated and remain the one for the previous cycle.

AFAIU the quorum is not outdated in this particular case. Because I think it should use the old quorum itself. Imagine `x` is the proposal that has not been flushed in the next  proposing period. When it is flushed, if we recompute the quorum, then it might be possible that the participation it recived has increased the quorum, and now the votes it recieved is no longer enough to meet the quorum. 

So basically, I think the quorum at the cycle in which the proposal was raised need to be used to evaluate it.

But I think this raise a case to actually store the quorum in the proposals itself. Or else in your example, if someone raise a proposal before `x` is flushed, then also the new quorum will come into play, and might stop `x` from getting passed.

Reply:
Have changed the proposals to include current quorum threhsolds and use them in quorum threshold checks.